### PR TITLE
Add icons to ProofSuggestions

### PR DESCRIPTION
**Pull request by mlsteele**
- [x] Requires https://github.com/keybase/keybase/pull/3432

### CODE REVIEW

**Comment by mlsteele** [comment]
Ready

File: go/externals/icon.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/keybase/client/go/libkb"
+	"github.com/keybase/client/go/protocol/keybase1"
+)
+
+func iconKeyMangle(key string) string {
```
Comment (joshblum) [inline go/externals/icon.go]: `normalizeIconKey`

File: go/externals/icon.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	case "http", "https", "dns":
+		return "web"
+	default:
+		return key
+	}
+}
+
+func ServiceHasFullIcon(key string) bool {
+	switch iconKeyMangle(key) {
+	case "btc", "facebook", "github", "hackernews", "pgp", "reddit", "rooter",
```
Comment (joshblum) [inline go/externals/icon.go]: could we detect if we have a parameterized config instead of hardcode list?

**Comment by mlsteele** [comment]
I think the only failure was `TestRunnerHandlePushBatch` which is unrelated and passes locally. Skipping

**Pull request merged**